### PR TITLE
Add fuzz tests for the network message decoder

### PR DESCRIPTION
**Pull request by ljedrz**
This PR adds a new workspace, `fuzz`, initialized and set up as per the `cargo-fuzz` [docs](https://rust-fuzz.github.io/book/cargo-fuzz/tutorial.html). It doesn't interfere with the snarkOS workspace.

I've been running it locally for a good amount of time and detected no issues so far.

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[package]
+name = "snarkos-fuzz"
+version = "0.0.0"
```
Comment (niklaslong) [inline fuzz/Cargo.toml:3]: Do we want to match the snarkOS version here?

File: fuzz/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+authors = ["Automatically generated"]
+publish = false
+edition = "2018"
```
Comment (niklaslong) [inline fuzz/Cargo.toml]: Why not `2021`?

**Review by niklaslong — APPROVED** [review]
Left a few comments but LGTM 👍

File: fuzz/Cargo.toml (hunk 1 above)
Comment (ljedrz) [reply fuzz/Cargo.toml:3]: this was auto-generated, I don't think it's intended to ever be published; if anything, I think we should rather add some README which would be updated with subsequent runs, stamped with the specific SHA of the repo (as the version is not updated as often)

File: fuzz/Cargo.toml (hunk 2 above)
Comment (ljedrz) [reply fuzz/Cargo.toml]: it was auto-generated; I guess I can bump it to 2021

File: fuzz/Cargo.toml (hunk 1 above)
Comment (niklaslong) [reply fuzz/Cargo.toml:3]: Perfect—so long as we keep track of what code corresponds to which run 👍

**Comment by ljedrz** [comment]
Added `Cargo.lock` to `.gitignore`, as it's not needed.

**Comment by ljedrz** [comment]
Added a second fuzz target that specifically targets deserialization (as opposed to the network message decoder).

**Comment by ljedrz** [comment]
Rebased.

**Review by howardwu — APPROVED** [review]

**Pull request merged**